# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-cache.c
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-cache.c
--- a/gfx/cairo/cairo/src/cairo-cache.c	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-cache.c	Mon May 06 19:25:12 2024 +0000
@@ -56,7 +56,6 @@
  * @keys_equal: a function to return %TRUE if two keys are equal
  * @entry_destroy: destroy notifier for cache entries
  * @max_size: the maximum size for this cache
- * Returns: the newly created #cairo_cache_t
  *
  * Creates a new cache using the keys_equal() function to determine
  * the equality of entries.
@@ -84,6 +83,8 @@
  * _cairo_cache_freeze() and _cairo_cache_thaw() calls can be
  * used to establish a window during which no automatic removal of
  * entries will occur.
+ *
+ * Returns: the newly created #cairo_cache_t
  **/
 cairo_status_t
 _cairo_cache_init (cairo_cache_t		*cache,
@@ -315,18 +316,18 @@
 			       closure);
 }
 
-unsigned long
+uintptr_t
 _cairo_hash_string (const char *c)
 {
     /* This is the djb2 hash. */
-    unsigned long hash = _CAIRO_HASH_INIT_VALUE;
+    uintptr_t hash = _CAIRO_HASH_INIT_VALUE;
     while (c && *c)
 	hash = ((hash << 5) + hash) + *c++;
     return hash;
 }
 
-unsigned long
-_cairo_hash_bytes (unsigned long hash,
+uintptr_t
+_cairo_hash_bytes (uintptr_t hash,
 		   const void *ptr,
 		   unsigned int length)
 {
@@ -336,3 +337,10 @@
 	hash = ((hash << 5) + hash) + *bytes++;
     return hash;
 }
+
+uintptr_t
+_cairo_hash_uintptr (uintptr_t hash,
+                     uintptr_t u)
+{
+    return _cairo_hash_bytes (hash, &u, sizeof(u));
+}